### PR TITLE
Add another test mode

### PR DESCRIPTION
**Pull request by GrumpyZhou**
Hi @kmyid,

I this pull request, i cleaned a bit the code and mainly provide another test mode -- test_simple, to have a clean output of test. 

You can try to test it under your env:
Run:
````
./main.py --run_mode=test_simple --res_dir="./" --log_dir="models" --test_log_dir="results"  --data_va="reichstag" --data_te="reichstag"
````
Outputs:
````
...
Loading valid data
Loading test data
Restoring from ./models/models-best...
Wed Aug 21 10:15:01 2019 Start testing on reichstag 
Finished computing essential matrix by network, samples 52, median/mean time per sample: 0.0088/0.0443s
>>>> Method ours Sample 52
Median Errs: err_q=0.0862 err_t=0.1378 inlier_ratio=0.1685 time=0.0016 
Mean AP [>deg,q,t]: 
(>5,0.5192, 0.2692)  (>10,0.6827, 0.4712)  (>15,0.7436, 0.5833)  (>20,0.7788, 0.6490)  

>>>> Method ours_ransac Sample 52
Median Errs: err_q=0.0869 err_t=0.1754 inlier_ratio=0.1030 time=0.0107 
Mean AP [>deg,q,t]: 
(>5,0.5192, 0.3269)  (>10,0.5962, 0.4135)  (>15,0.6923, 0.4936)  (>20,0.7644, 0.5529)  

>>>> Method ransac Sample 52
Median Errs: err_q=0.3884 err_t=0.4594 inlier_ratio=0.1382 time=0.2864 
Mean AP [>deg,q,t]: 
(>5,0.1538, 0.0769)  (>10,0.2596, 0.1442)  (>15,0.3205, 0.1859)  (>20,0.3606, 0.2548)  

Save final results to  results/reichstag_test_res.npy
````
Evaluation results now is saved compactly inside reichstag_test_res.npy.

### CODE REVIEW

**Comment by kmyid** [comment]
Thank you for the PR. @jyh2005xx Can you check if merging this will conflict with what you have done for another project?  @weiweisun2018 Can you have a look at the PR when you have time ?

**Comment by weiweisun2018** [comment]
I have checked this PR. It provides a clean output for testing, which is good. For example, it saves the output as a unified .npy file instead of .txt files.  It makes the output cleaner. But in my opinion, it is not very necessary if you want to simply check the numbers in the Linux system or in server. I personally would stick to the .txt files for now.

**Comment by GrumpyZhou** [comment]
Yes as @weiweisun2018  said, it is not a must PR. It just provides ANOTHER easier mode for testing, while old mode to produce txt results remains. If someone else wants to use your code, my code might help them to understand quicker details what the codes are doing, since it took me several hours to check line by line for data loading and testing code. 

On the other hand, i am continuing simplifying things since i need to further some research based on your codes. But might most of them are not NECESSARY, since they are just more user-friendly for readability and reusability, while not changing the core functioning (and they shouldn't right? :) ).

In conclusion, i don't mind not contributing at all. If you don't want these changes, i will just close this PR and also not adding more PR to this original PR.

**Comment by kmyid** [comment]
I think it would be cool to have. I really appreciate these PRs. Especially because this code release was rushed and very dirty!

One thing though is that it would be nice to not create the npy output by default and have it as an opt-in. Would that be something that we could arrange?

Again, thx for all the time and effort. And also for sharing your stuff!!

**Comment by GrumpyZhou** [comment]
@kmyid , yes i can do that. You mean add it in config as option? By the way, i just noticed that you might doesn't want commit  d010a5f. So shall i close this pull request and open another one to leave out your unwanted commits (if there's more just tell me) and the new feature you want ?

**Comment by GrumpyZhou** [comment]
@kmyid  

I updated the PR,  now by default it doesn't save things. To save the output, run:
````
./main.py --run_mode=test_simple --res_dir="./" --log_dir="models" --test_log_dir="results"  --data_va="reichstag" --data_te="reichstag" --save_res_npy
````

I also remove the commit that rm the datasets/.placeholder .

**Comment by kmyid** [comment]
Sounds good! I'll have one final look probably next Monday and merge :-)
